tiledb/sm/c_api/tiledb.cc

```diff
@@ -455,64 +456,51 @@ int32_t tiledb_array_schema_load(
     const char* array_uri,
     tiledb_array_schema_t** array_schema) {
   // Create array schema
+  ensure_context_is_valid(ctx);
+  ensure_output_pointer_is_valid(array_schema);
   *array_schema = new (std::nothrow) tiledb_array_schema_t;
   if (*array_schema == nullptr) {
-    auto st = Status_Error("Failed to allocate TileDB array schema object");
-    LOG_STATUS_NO_RETURN_VALUE(st);
-    save_error(ctx, st);
-    return TILEDB_OOM;
+    throw CAPIStatusException("Failed to allocate TileDB array schema object");
```

Comment:
You can throw `bad_alloc`, which will map to `TILEDB_OOM` in the C API.

Reply:
Done